src/app/GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs

```diff
@@ -484,7 +484,15 @@ public void LoadingCompleted()
 
                 if (_toBeSelectedGraphIndexesCache.Value.Count == 0)
                 {
-                    // Nothing to select or interrupted
+                    // Nothing to select or interrupted, select the first shown real commit or the first row (which exists here)
```

Comment:
This check should probably be further below too, should _loadedToBeSelectedRevisionsCount that should be checked too?